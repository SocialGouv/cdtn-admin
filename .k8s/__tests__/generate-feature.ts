//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  expect(
    await getEnvManifests("dev", "", {
      ...project("cdtn-admin").dev,
      //
      // Feature like env
      CI_COMMIT_REF_NAME: "jean-mich/Jeune-homme-à-la-roses-1.2.3",
      CI_COMMIT_REF_SLUG: "jean-mich-jeune-homme-à-la-roses-1-2-3",
      CI_ENVIRONMENT_NAME: "jean-mich/Jeune-homme-à-la-roses-1.2.3",
      CI_ENVIRONMENT_SLUG: "jean-mich-jeune-123456",
      CI_ENVIRONMENT_URL: `https://jean-mich-jeune-123456-cdtn-admin.dev2.fabrique.social.gouv.fr`,
      KUBE_NAMESPACE: `cdtn-admin-85-jean-mich-jeune-123456`,
      PRODUCTION: "",
      //
      RANCHER_PROJECT_ID: "c-bar:p-foo",
    })
  ).toMatchSnapshot();
});
