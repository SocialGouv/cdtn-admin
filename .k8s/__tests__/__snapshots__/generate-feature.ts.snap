// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-bar:p-foo
    git/branch: jean-mich/Jeune-homme-à-la-roses-1.2.3
    git/remote: >-
      https://gitlab-ci-token:[MASKED]@gitlab.factory.social.gouv.fr/SocialGouv/cdtn-admin.git
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    azure-pg-admin-user: cdtn-admin
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: cdtn-admin-85-jean-mich-jeune-123456
---
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    app: alert
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: alert
spec:
  concurrencyPolicy: Forbid
  successfulJobsHistoryLimit: 3
  failedJobsHistoryLimit: 3
  schedule: 0 1 * * *
  jobTemplate:
    spec:
      backoffLimit: 0
      template:
        spec:
          containers:
            - name: update-alert
              image: >-
                registry.gitlab.factory.social.gouv.fr/socialgouv/cdtn-admin/alert:0123456789abcdefghijklmnopqrstuvwxyz0123
              resources:
                requests:
                  cpu: 1500m
                  memory: 2.5Gi
                limits:
                  cpu: 2000m
                  memory: 3Gi
              workingDir: /app
              env:
                - name: PRODUCTION
                  value: ''
                - name: HASURA_GRAPHQL_ENDPOINT
                  value: http://hasura/v1/graphql
              envFrom:
                - secretRef:
                    name: cdtn-admin-secrets
              volumeMounts:
                - name: tz-paris
                  mountPath: /etc/localtime
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
            fsGroup: 1000
            supplementalGroups:
              - 1000
          volumes:
            - name: tz-paris
              hostPath:
                path: /usr/share/zoneinfo/Europe/Paris
          restartPolicy: Never
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    app: hasura
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: hasura
  namespace: cdtn-admin-85-jean-mich-jeune-123456
spec:
  replicas: 1
  selector:
    matchLabels:
      app: hasura
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-cdtn-admin
        app.gitlab.com/env: jean-mich-jeune-123456
        app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
      labels:
        app: hasura
        application: jean-mich-jeune-123456-cdtn-admin
        owner: cdtn-admin
        team: cdtn-admin
    spec:
      containers:
        - image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/cdtn-admin/hasura:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: hasura
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 1000m
              memory: 1.5Gi
            requests:
              cpu: 100m
              memory: 64Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: azure-pg-user-jeanmichjeune123456
            - secretRef:
                name: hasura
            - configMapRef:
                name: hasura
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user-jeanmichjeune123456
          image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/docker/wait-for-postgres:5.0.1
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  name: hasura
  namespace: cdtn-admin-85-jean-mich-jeune-123456
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
spec:
  encryptedData:
    HASURA_GRAPHQL_ADMIN_SECRET: >-
      AgCDbpjKjQMq/294VjVxUwVk2OytCAXcaX8PhppBTPozstmZvvhKvszeKF/Ki0qyM1u0k29nBZ86iKJ63VV5wCEKs1l4Kyv3YKNdUpyb4L8zKktfR9sY6UlPypBRfTs/aZXO4FCnTJ4Jsxyaiyynumxuw0oTvLGQoio8IKTg2d7eGu4IRiOLpaLD8GXTmYBuCSTs5cT1y4pBXVu+iJXsLDL4OApYeJUG4xTLtZiBRXDtU316NKpEkvmLTZRkDSoqTBVlOlYLwamYDwyoOuoRPuziqIhjYH7uWSB6QbkF3+AknckfoCCrTaeLitemmXLgvzK6Yxbzr1OOb0DaE3lmpRMOPHpANy5BFYCC+j8Iu02y/A6p1Is793OujN+0m2+Np5eXKsOmKHMbSIQjvtRHh+eQIT5cEqXsZoIsrAVOE+fZn6PDsSZJkFtm+yb4+7dspMdm1VsyKl7VIUS2+U4aex/IsPzvzthbRpK6iK3k5fIiCvSZ65yhG0sxV9T6kxg6+lM5tV3Y3cE5NeKPiFjhhASBfDpkKpYyGI7DuAiLFXzHNm/DE4rarJ1Kkh5tKriSyqMAsyhbR0b8KESNwUjoLRUY8ThS3P6xhkobx0CEdxqKgzZV3dTeboxo34TfSij0eCGnTCInd9vj8X6y7qmRiqMWGZ1uQTPqRzWkj+cC3Op+UhVpnyqr+GpFI4J7Ru2g/Z8my1vbV5fXYcs=
    HASURA_GRAPHQL_JWT_SECRET: >-
      AgDSW4ceV9HFoxBNqg/hGa6uunl76cjtKBRSYf5LEPKzNHtXixd73LoFGa2y6/cQ/yGevNU6k7w62pkbWnz5rNAznsBE8QtBt1eRKQqjyJJ21m4sWI/MK6UI3gUS/Ad+LeCMxSoKG3XEzFSAwSndSXsgmwAP4KPpBF83DvdpVVpZtAjvPbNYNE6Ocjhtmt/TLqWjsV4Z+hkS8CKGk+PsYDEuR74Mj5TJys0MALJA9TB2dTIqMHOVZofD8T9hKH42yJWCDruH+gNNPKcleG/R0Pt7LJSamZtsqtXdxeE02tB+I9/VaEMrBL2WCJKhGHTjlDxexl4V/fswFYgXc/T4+hmPVB76xWVi6yFfggD/tLcAvgw3Ivdmbk/1c4CayKjU0+B+57EVrRqCGbJieBfbv6L3EWjU0uQj/uJEaZxIF4st3/r+iSGnrtPveBqnUgkJeqOmhf4sWJwE3xsPJM2kNYDPelWpYdxpoPaq3PNcLs7WpxMtRvwvvz2LMsWhEuUyHpe5yLOvi5swhcGqX1ALxGxfxzV+O229WfuTmV/kGUmcHhdf/VXdJRa+PYS83YK39wC1ZsoXJvPjAvEMgoTaRE8NHPWVD2ookNwdiRQIoLrSrGMP6F+Team2KtCic9ONYnM2wwUj2JCQRkFX2LZXwm9g9uR/Leuyd/IL5+tczwfF+PHfs6TQv+uKVz2qoZQZk4myQ/0JwHluOSXHTwEWkvuwGYO8QMTDOTTdp4Dor3kN7hQZyvWf41V/EHmOoLeFsdpQkWexo1FWHRZYtgG7ZVrgs07bY65gY6c1tZOKFaWhacTuneiLosz79vxn5omsFpjOOiWdSQOPZU8GkGxqJVq16H3VJg==
    PUBLICATION_SECRET: >-
      AgAtDPOTr9khd47wV5WPXJaLSj+6Kmh5N/Wt9SJUTmRgn9jp8trclhS5DvZxmfVv/rYRRiDTuPATDR4yMbfvtrO9ABAhmRZFvmU5VbxdtCDrc9qtv5ZJ4dsYIVxyXIad7+CumHkunXYSwgmCqr2bvv5oeWRmTx2x4uwaNzhEKQ4hUzYCyWp9laJHt5mYT/c/IWZu0PaDXJQVWQuep3fJvjtxQw9EZVy+ehGAqyClL7bkqT1KVrVBDfvpPeO97O2e2de49RcdZSCtBTmGDHKg7fwUziBJSSI8ObtUMzWm56223XZXKqrxdChAlWJE05tHsvowa18NadFDjJwVt3RtPp+HguVj+I7PiTJMTjyZpsmHlr767teLYrXAoAVc9MRrGeznvYw4i0DhNFZXfyYghSKnKPW4yfYzNKwX50huCyQethn1TqU6MAYGrWiyx9XsaHy3bNgAO4ru9o8WV7bfyeRRsABMoiubyq9RCxfOXRU3cbWYaiS8aiPxALZRKjhSeQhJ5ufdL700rJWV2nuoxXG8jjR3yw3bpCL6imIrLRIAP1+Wki5I8Qo98dfBBF/nahEWeXdCqCjpImfSLUioKMes2IrH8ragfja+85x06S8717b67qyBqtGeGAfX2JYLDWILoA0k9RdzyIwuakAZkChGoWTUy8aRjC7GJIKCQ5IKZ5FQr0IuVVjj1xoGBh6O/sFsOLF7/sA3hRcvirzcCFvCIwsmHs/q4K1iU7nDR0WE1AUV/uccbP3yzAzv6gqElztn1f/LOpA01b2Wn9YAzi6S65wUXUWx7UAx6YaiiofmnySnseSNfE+8
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-cdtn-admin
        app.gitlab.com/env: jean-mich-jeune-123456
        app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
      name: www
      labels:
        application: jean-mich-jeune-123456-cdtn-admin
        owner: cdtn-admin
        team: cdtn-admin
    type: Opaque
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: hasura
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  namespace: cdtn-admin-85-jean-mich-jeune-123456
data:
  ACCOUNT_EMAIL_WEBHOOK_URL: http://www/api/webhooks/account
  API_URL: http://www/api
  HASURA_GRAPHQL_ENABLE_CONSOLE: 'false'
  HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
  HASURA_GRAPHQL_NO_OF_RETRIES: '5'
  HASURA_GRAPHQL_SERVER_PORT: '80'
  HASURA_GRAPHQL_UNAUTHORIZED_ROLE: public
  PUBLICATION_WEBHOOK_URL: http://www/api/webhooks/publication
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: hasura
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: hasura
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  namespace: cdtn-admin-85-jean-mich-jeune-123456
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: hasura
  type: ClusterIP
---
apiVersion: batch/v1beta1
kind: CronJob
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    app: ingester
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: ingester
spec:
  concurrencyPolicy: Forbid
  successfulJobsHistoryLimit: 3
  failedJobsHistoryLimit: 3
  schedule: 30 0 * * *
  jobTemplate:
    spec:
      backoffLimit: 0
      template:
        spec:
          containers:
            - name: update-ingester
              image: >-
                registry.gitlab.factory.social.gouv.fr/socialgouv/cdtn-admin/ingester:0123456789abcdefghijklmnopqrstuvwxyz0123
              resources:
                requests:
                  cpu: 1500m
                  memory: 2.5Gi
                limits:
                  cpu: 2000m
                  memory: 3Gi
              workingDir: /app
              env:
                - name: HASURA_GRAPHQL_ENDPOINT
                  value: http://hasura/v1/graphql
              envFrom:
                - secretRef:
                    name: azure-pg-user-jeanmichjeune123456
              volumeMounts:
                - name: data
                  mountPath: /app/data
                - name: tz-paris
                  mountPath: /etc/localtime
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
            fsGroup: 1000
            supplementalGroups:
              - 1000
          volumes:
            - name: data
              persistentVolumeClaim:
                claimName: ingester
            - name: tz-paris
              hostPath:
                path: /usr/share/zoneinfo/Europe/Paris
          restartPolicy: Never
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: ingester
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 2Gi
---
apiVersion: batch/v1
kind: Job
spec:
  backoffLimit: 0
  template:
    spec:
      containers:
        - command:
            - create-db-user
          env:
            - name: NEW_DB_NAME
              value: autodevops_jeanmichjeune123456
            - name: NEW_USER
              value: user_jeanmichjeune123456
            - name: NEW_PASSWORD
              value: password_jeanmichjeune123456
            - name: NEW_DB_EXTENSIONS
              value: hstore pgcrypto citext uuid-ossp
          envFrom:
            - secretRef:
                name: azure-pg-admin-user
          image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:2.6.1
          imagePullPolicy: IfNotPresent
          name: create-db-user
          resources:
            limits:
              cpu: 300m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 64Mi
      restartPolicy: Never
    metadata:
      annotations:
        app.gitlab.com/app: socialgouv-cdtn-admin
        app.gitlab.com/env: jean-mich-jeune-123456
        app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
      labels:
        application: jean-mich-jeune-123456-cdtn-admin
        owner: cdtn-admin
        team: cdtn-admin
  ttlSecondsAfterFinished: 86400
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: create-db-job-jeanmichjeune123456
  namespace: cdtn-admin-85-jean-mich-jeune-123456
---
apiVersion: v1
kind: Secret
stringData:
  DATABASE_URL: >-
    postgresql://user_jeanmichjeune123456%40cdtnadmindevserver.postgres.database.azure.com:password_jeanmichjeune123456@cdtnadmindevserver.postgres.database.azure.com/autodevops_jeanmichjeune123456?sslmode=require
  DB_URI: >-
    postgresql://user_jeanmichjeune123456%40cdtnadmindevserver.postgres.database.azure.com:password_jeanmichjeune123456@cdtnadmindevserver.postgres.database.azure.com/autodevops_jeanmichjeune123456?sslmode=require
  HASURA_GRAPHQL_DATABASE_URL: >-
    postgresql://user_jeanmichjeune123456%40cdtnadmindevserver.postgres.database.azure.com:password_jeanmichjeune123456@cdtnadmindevserver.postgres.database.azure.com/autodevops_jeanmichjeune123456?sslmode=require
  PGDATABASE: autodevops_jeanmichjeune123456
  PGHOST: cdtnadmindevserver.postgres.database.azure.com
  PGPASSWORD: password_jeanmichjeune123456
  PGRST_DB_URI: >-
    postgresql://user_jeanmichjeune123456%40cdtnadmindevserver.postgres.database.azure.com:password_jeanmichjeune123456@cdtnadmindevserver.postgres.database.azure.com/autodevops_jeanmichjeune123456?sslmode=require
  PGSSLMODE: require
  PGUSER: user_jeanmichjeune123456@cdtnadmindevserver.postgres.database.azure.com
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: azure-pg-user-jeanmichjeune123456
  namespace: cdtn-admin-85-jean-mich-jeune-123456
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    app: www
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: www
  namespace: cdtn-admin-85-jean-mich-jeune-123456
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-cdtn-admin
        app.gitlab.com/env: jean-mich-jeune-123456
        app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
      labels:
        app: www
        application: jean-mich-jeune-123456-cdtn-admin
        owner: cdtn-admin
        team: cdtn-admin
    spec:
      containers:
        - image: >-
            registry.gitlab.factory.social.gouv.fr/socialgouv/cdtn-admin/www:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 1000m
              memory: 560Mi
            requests:
              cpu: 5m
              memory: 560Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          env:
            - name: COMMIT
              value: 0123456789abcdefghijklmnopqrstuvwxyz0123
            - name: FRONTEND_HOST
              value: >-
                https://jean-mich-jeune-123456-cdtn-admin.dev2.fabrique.social.gouv.fr
            - name: VERSION
              value: jean-mich/Jeune-homme-à-la-roses-1.2.3
          envFrom:
            - secretRef:
                name: www
            - configMapRef:
                name: www
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  name: www
  namespace: cdtn-admin-85-jean-mich-jeune-123456
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
spec:
  encryptedData:
    AZURE_STORAGE_ACCOUNT_KEY: >-
      AgDQCgdeUi0yKlErFvxzFh0gxaoi3pX6EVHnUAIYikjD0krf7YrtUNo2RrdneBVrvbeR4F1xFiQLN+Km80NsNzyCNIgPjsV0EALr0hx7TEg42XUpyvC2aH0QpbF7Zj+Xo4y9vQI/DdJIBDQ/Oo9btABzjPa6anOu0LCP2PIz5B2ndUGnlWSaV1JtArK7jOEA5tUeaSXGXdfjOj4JnVkJKgFCdihWmt8lQMI5VAqqKB3pbAL0c/45jI3UVxqmiL4pa0iJwh3fPa1SBZCeu7IWSRYzC46d3hvmBD6IcEFejS5uU+if6JizzSH4sVQQ0A2lTiIj67wLwmJ2LgB2vnm/bCeDMtzmOUI8PfsOUPj94L62r7kZbaHAZz3uzeUn/K6HlMnD7cDI75nrtSB2o330cK2wTc9yrPU8Td5k5niZObHvbyCYy/nQPDRHdwxUsH0YOkGHyGNyRBggh7/y1Hzp+NKqV/eCq+RKhS+g7NOdDBaGjEXRZ7FGmpyacyGZLbFORtgX1Xn0f1HC/I7nVYys1RoV2RJiWo0K3LwuNA/RW5RpA8rjfXx2/HPcTXMqrXD/QxX9pl75sAB5mOsjBFye5MGw18W73gZ2uazEp8RoxuLn/6iImtW7ah9XoYeP8iSlFDN+RY2hMv8X1SdIov6lqCUsBrwP7oUo6cKG9V77VQ/1IFUxXM+HgPQSFtJxNGPitp8x4+cLsjPfFdgnJ1NInx8WUhPwJQrwvmfpMGZO+OK9c5mVGjdro6JN/6e3lzAEKrO8nOeVPy6hBn+KEmxBV+m4oA9DU3RgsoapIySnnGmihG+UBEWYWVk6
    AZURE_STORAGE_ACCOUNT_NAME: >-
      AgCtRuQPvQKRuXlApVMqRhNZTsT7Vh/pNdJkEcG0Ox24Fll4RfeKFJPaazspdTq6gM3f6MqEZqoTgvsOg/Jez3rLDaSPcgwetH7bC7Ut3gEzBeJQpO+phCxByxQU2XQfxYQwxGtMrsCfN6h9MalETyg8AnCJoLP4kyKDuyLjo0Hvgwr3XokmV38CfPRDwWYvvBCED4EwsuCO1LvIOcC7XnlGEnrA9p+6T8jEl1z86bd14+AbXAC4xuVOluvrPpgMEjrpI4jqkNuJDTt80W48hWH0ZayzAKQ69uSTKUe5011i8ui1/frD0CDNPpFoI/CQqgNB9wZYT6I2rMVl9sXWtMexq+y96/YuiaQyJKAGB9Foxq6JV2viVEpNHqf2egw7RhWnJSPe0XJqFFyHkUz4FA+dDhL5AcIFdknwlvjzM9VkSj6todKucml6hUdorQ0GZYFy5+QkA5OqUxn45bcW2AVLYYSTlWMRTmjl0HTPTnILDsfv6nml0cG+rHmVu4rWjTT8rPoQRCAIeNKg4CxCzBPojhdq49jTvmy8lG8lSV1qdL8ZDipCja3+LIiQztS4qu0pZXIKp3smfhCWI5ENXVQ8UIZCthTac+Yg2vsl0jaOKaGP4SdXCxPZTdgqclGjm8BUSP3ZdDLbgl8Z6wqxsXN8RKsI/kCaURu4Ner1YqJnFmX4qiHZsnVit3GNpo6Qiv3Ogqr2zpEbInWyML0=
    PUBLICATION_SECRET: >-
      AgAtDPOTr9khd47wV5WPXJaLSj+6Kmh5N/Wt9SJUTmRgn9jp8trclhS5DvZxmfVv/rYRRiDTuPATDR4yMbfvtrO9ABAhmRZFvmU5VbxdtCDrc9qtv5ZJ4dsYIVxyXIad7+CumHkunXYSwgmCqr2bvv5oeWRmTx2x4uwaNzhEKQ4hUzYCyWp9laJHt5mYT/c/IWZu0PaDXJQVWQuep3fJvjtxQw9EZVy+ehGAqyClL7bkqT1KVrVBDfvpPeO97O2e2de49RcdZSCtBTmGDHKg7fwUziBJSSI8ObtUMzWm56223XZXKqrxdChAlWJE05tHsvowa18NadFDjJwVt3RtPp+HguVj+I7PiTJMTjyZpsmHlr767teLYrXAoAVc9MRrGeznvYw4i0DhNFZXfyYghSKnKPW4yfYzNKwX50huCyQethn1TqU6MAYGrWiyx9XsaHy3bNgAO4ru9o8WV7bfyeRRsABMoiubyq9RCxfOXRU3cbWYaiS8aiPxALZRKjhSeQhJ5ufdL700rJWV2nuoxXG8jjR3yw3bpCL6imIrLRIAP1+Wki5I8Qo98dfBBF/nahEWeXdCqCjpImfSLUioKMes2IrH8ragfja+85x06S8717b67qyBqtGeGAfX2JYLDWILoA0k9RdzyIwuakAZkChGoWTUy8aRjC7GJIKCQ5IKZ5FQr0IuVVjj1xoGBh6O/sFsOLF7/sA3hRcvirzcCFvCIwsmHs/q4K1iU7nDR0WE1AUV/uccbP3yzAzv6gqElztn1f/LOpA01b2Wn9YAzi6S65wUXUWx7UAx6YaiiofmnySnseSNfE+8
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-cdtn-admin
        app.gitlab.com/env: jean-mich-jeune-123456
        app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
      name: www
      labels:
        application: jean-mich-jeune-123456-cdtn-admin
        owner: cdtn-admin
        team: cdtn-admin
    type: Opaque
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  namespace: cdtn-admin-85-jean-mich-jeune-123456
data:
  ACCOUNT_MAIL_SENDER: contact@fabrique.social.gouv.fr
  GITLAB_PROJECT_ID: '136'
  GITLAB_URL: https://gitlab.factory.social.gouv.fr/api/v4
  HASURA_GRAPHQL_ENDPOINT: http://hasura/v1/graphql
  JWT_TOKEN_EXPIRES: '15'
  NEXT_PUBLIC_ACTIVATION_TOKEN_EXPIRES: '10080'
  NEXT_PUBLIC_REFRESH_TOKEN_EXPIRES: '43200'
  NODE_ENV: production
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  namespace: cdtn-admin-85-jean-mich-jeune-123456
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: www
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: jean-mich-jeune-123456
    app.gitlab.com/env.name: jean-mich/Jeune-homme-à-la-roses-1.2.3
  labels:
    app: www
    application: jean-mich-jeune-123456-cdtn-admin
    owner: cdtn-admin
    team: cdtn-admin
  name: www
  namespace: cdtn-admin-85-jean-mich-jeune-123456
spec:
  rules:
    - host: jean-mich-jeune-123456-cdtn-admin.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: www
              servicePort: 80
            path: /
  tls:
    - hosts:
        - jean-mich-jeune-123456-cdtn-admin.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;
